src/Chapter3.hs

```diff
@@ -970,7 +1053,21 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Weekday = Sat | Sun | Mon | Tue | Wed | Thu | Fri deriving (Show, Eq, Enum, Bounded)
+
+isWeekend :: Weekday -> Bool
+isWeekend weekday = case weekday of
+  Sat -> True
+  Sun -> True
+  _ -> False
+
+nextDay :: Weekday -> Weekday
+nextDay weekday
+  | weekday == maxBound = minBound
+  | otherwise = succ weekday
 
+daysToParty :: Weekday -> Int
+daysToParty weekday = fromEnum Fri - fromEnum weekday
```

Comment:
This would show the shortest distance to `Friday`, however, time doesn't work like this.
For example. if it is `Saturday`, it would tell you that it is -1 day to `Friday`, however, it is 6 days to go 🥳

Reply:
I changed the regular order of days because of this, but I am aware that this is a cheat :cold_sweat:

Reply:
Yeah, this one way to do it, but I agree, that this looks like a cheat 😄 

You can apply the knowledge that we know exactly that there are 7 days in a week. So we can just calculate everything by `mod` 7!

Let's look how it would work:

```haskell
ghci> mod (5) 7
5
ghci> mod (-1) 7
6
```

Looks like just what we need!